src/TableFunctions/TableFunctionFile.cpp

```diff
@@ -26,6 +26,8 @@ void TableFunctionFile::parseFirstArguments(const ASTPtr & arg, const ContextPtr
     {
         ITableFunctionFileLike::parseFirstArguments(arg, context);
         StorageFile::parseFileSource(std::move(filename), filename, path_to_archive);
+        for (auto & file_path : filenames)
+            StorageFile::parseFileSource(std::move(file_path), file_path, path_to_archive);
```

Comment:
In `StorageFile::parseFileSource()` we separate the name of the archive from the name of the file. But in our case, we will only get the `path_to_archive` of the last entry. It seems like to support archives fully we should save all the paths to archives to use them afterwards.

Reply:
We should also add tests with archives.